golang/src/queries/users.go

```diff
@@ -0,0 +1,27 @@
+package queries
+
+import "database/sql"
+
+// IsUser returns whether the provided user exists in the database
+func IsUser(db *sql.DB, username string) (bool, error) {
+	var (
+		count int64
+		query = `SELECT COUNT(*) FROM ( SELECT DISTINCT id FROM users WHERE username = $1 ) AS check_user`
```

Comment:
for the benefit of other reviewers: because this is code/DB stuff I'm unfamiliar with I checked -- this username field is the long username e.g. `mian@iplantcollaborative.org` rather than `mian`; this is the username which terrain (the only client of user-sessions) passes along, so this is correct.